src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeCleanupFixer.cs

```diff
@@ -320,37 +340,138 @@ private async Task<bool> FixTextBufferAsync(TextBufferCodeCleanUpScope textBuffe
                         }
                     });
 
-                    var document = buffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
-                    var codeCleanupService = document.GetLanguageService<ICodeCleanupService>();
+                    var newSolution = await FixSolutionAsync(solution, context.EnabledFixIds, progressTracker, cancellationToken).ConfigureAwait(true);
 
-                    var allDiagnostics = codeCleanupService.GetAllDiagnostics();
+                    await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
I thought this whole thing can only be called from UI thread?

Reply:
the above ConfigureAwait is using true, so it should be in UI thread?

Reply:
Asynchronous methods are not allowed to assume the caller is on a particular thread.

Reply:
I verified that `SwitchToMainThreadAsync` is not called redundantly. 👍